File: public/cses/introductory-problems/missing-number/sum-to-n-approach/missing-number.cpp
#include <bits/stdc++.h>
using namespace std;

// setup-start
void setup() {
#ifndef ONLINE_JUDGE
    // For getting input from input.txt file
    freopen("../input.txt", "r", stdin);

    // Prlong longing the Output to output.txt file
    freopen("../output.txt", "w", stdout);
#endif
}
// setup-end

int main() {
    setup();

    int n;
    cin >> n;

    long long total = 0;
    for (int i = 0; i < n - 1; i++) {
        int v;
        cin >> v;
        total += v;
    }
    long long sum_upto_n = (1LL * n * (n + 1)) / 2;

    int res = sum_upto_n - total;
    cout << res << endl;
}

File: public/cses/introductory-problems/missing-number/sum-to-n-approach/missing-number.py
# setup-start
def setup():
    import os
    import sys
    script_directory = os.path.dirname(os.path.realpath(__file__))
    parent_directory = os.path.dirname(script_directory)
    input_path = os.path.join(parent_directory, 'input.txt')
    output_path = os.path.join(parent_directory, 'output.txt')
    if os.path.exists(input_path):
        sys.stdin = open(input_path, 'r')
        try:
            sys.stdout = open(output_path, 'w')
        except Exception as e:
            pass
# setup-end


def main():
    setup()
    n = int(input())
    elements = [int(v) for v in input().split()]
    sum_upto_n = (n * (n + 1)) // 2
    res = sum_upto_n - sum(elements)
    print(res)


main()
